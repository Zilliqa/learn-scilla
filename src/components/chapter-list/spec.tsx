import React from 'react';
import ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import intructionsLocalized from '../../course/instructions/en/index';
import ChapterList from '.';

const t = (s: string) => s;

describe('Chapter Complete Card tests', () => {
  const baseComponent = (props) => (
    <MemoryRouter initialEntries={['/']}>
      <ChapterList
        chapterList={intructionsLocalized}
        isAuth={props.isAuth}
        progress={props.progress}
        t={t}
      />
    </MemoryRouter>
  );

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const tree = renderer
        .create(baseComponent({ isAuth: true, progress: { chapter1: 3 } }))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders without crashing when auth is false', () => {
      const div = document.createElement('div');
      ReactDOM.render(baseComponent({ isAuth: false, progress: undefined }), div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders without crashing when auth is true', () => {
      const div = document.createElement('div');
      ReactDOM.render(baseComponent({ isAuth: true, progress: { chapter1: 3 } }), div);
      ReactDOM.unmountComponentAtNode(div);
    });
  });
});
