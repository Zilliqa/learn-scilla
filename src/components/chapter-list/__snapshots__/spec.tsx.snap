// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chapter Complete Card tests basic tests matches the snapshot 1`] = `
<div>
  <div
    className="m-2"
  >
    <a
      aria-label="start chapter"
      className="btn btn-primary btn-block text-left"
      href="/chapter/1/lesson/4"
      onClick={[Function]}
    >
      <small>
         
        chapter.chapter 1 :
      </small>
       Fundamentals 
      <small>
        (3/9)
      </small>
    </a>
  </div>
</div>
`;
